stock_storage_type/models/stock_storage_category_capacity.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2022 ACSONE SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StorageCategoryProductCapacity(models.Model):
+
+    _inherit = "stock.storage.category.capacity"
+
+    allow_new_product = fields.Selection(
+        [
+            ("empty", "If the location is empty"),
+            ("same", "If all products are same"),
+            ("mixed", "Allow mixed products"),
+        ],
+        default="mixed",
+        required=True,
+    )
+    do_not_mix_lots = fields.Boolean(
```

Comment:
This needs to be replaced by an additional option in the selection `allow_new_product` that you can name `same_lot` = "If all lots are the same"

Reply:
@jbaudoux I tried to do it but I don't remember the case where this does not work. I'll keep you in touch